src/tool/hpcrun/gpu/nvidia/cupti-api.c

```diff
@@ -1482,10 +1482,12 @@ cupti_activity_flush
 void
 cupti_device_flush(void *args)
 {
-  cupti_stop_flag_unset();
-  cupti_activity_flush();
-  // TODO(keren): replace cupti with sth. called device queue
-  gpu_application_thread_process_activities();
+  if (cupti_stop_flag) {
```

Comment:
If there's no stop test in `cupti_device_shutdown`, a program without gpu api invocation but is monitored by "hpcrun -e gpu" still causes a deadlock. @blue42u has a test case.

https://github.com/HPCToolkit/hpctoolkit/pull/196

In fact, originally I didn't test `cupti_stop_flag` in ` cupti_device_shutdown`, because I assumed a user always knows a programs uses GPU before he monitors.